src/WinRT.Runtime/ActivationFactory.cs

```diff
@@ -0,0 +1,289 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using WinRT.Interop;
+
+namespace WinRT
+{
+    internal unsafe sealed class DllModule
+    {
+        private static readonly string _currentModuleDirectory = AppContext.BaseDirectory;
+
+        private static readonly Dictionary<string, DllModule> _cache = new Dictionary<string, DllModule>(StringComparer.Ordinal);
+
+        private readonly string _fileName;
+        private readonly IntPtr _moduleHandle;
+        private readonly delegate* unmanaged[Stdcall]<IntPtr, IntPtr*, int> _GetActivationFactory;
+        private readonly delegate* unmanaged[Stdcall]<int> _CanUnloadNow; // TODO: Eventually periodically call
+
+        public static bool TryLoad(string fileName, out DllModule module)
+        {
+            lock (_cache)

```

Comment:
It is now more likely that we see parallel calls here due to different threads activating different types.  Given these are already cached too, it shouldn't have too much of an impact, but will need to monitor perf traces to see if this shows up and if we need to improve the locking here.  But for now, this is fine.